### PR TITLE
[mlir][sparse] Implement rewriters to reinterpret maps on foreach

### PR DESCRIPTION
**Pull request by PeimingLiu**

### CODE REVIEW

**Review by aartbik — APPROVED** [review]
Good to go after merging with the latest change to this file.

File: mlir/lib/Dialect/SparseTensor/Transforms/SparseReinterpretMap.cpp
Comment (yinying-lisa-li) [inline mlir/lib/Dialect/SparseTensor/Transforms/SparseReinterpretMap.cpp:217]: For consistency, maybe handle?

File: mlir/lib/Dialect/SparseTensor/Transforms/SparseReinterpretMap.cpp
Comment (yinying-lisa-li) [inline mlir/lib/Dialect/SparseTensor/Transforms/SparseReinterpretMap.cpp:268]: nit: .

**Pull request merged**
